frontend/src/pages/StudentDashboard.js

```diff
@@ -19,54 +19,69 @@ class StudentDashboard extends React.Component {
             class_standing: ''
         },
         hours: [],
+        approved_hours: '',
+        pending_hours: '',
         notifications: [],
     }
 
-
     componentDidMount() {
         getUserDataForStudent(this);
         getHoursForStudent(this);
     }
 
     render() {
+    //This will prevent the faculty or mentor from accessing the student page
+    let role = this.props.userData.role;
+        if (role === 2) {
+            return <Redirect to="/app/mentor" push/>
+        } else if (role === 0) {
+            return <Redirect to="/app/" push/>
+        }
         return <Container className="student-page">
             <Row area-label="top spacer">
 
             </Row>
             <Paper className="student-progress-profile">
                 <Row>
                     <Col>
-                        <UserPic imgSrc={this.state.userData.user.profile_picture}>
-
-                        </UserPic>
+                        <Container className="user-pic">
+                            <UserPic imgSrc={this.state.userData.user.profile_picture}/>
+                        </Container>
                     </Col>
                     <Col>
-                        Welcome
-                        Back, {this.state.userData.user.first_name} {this.state.userData.user.last_name} - {this.state.userData.student_id}!
+                        <Row className="welcome-text">
+                            Welcome
+                            Back, {this.state.userData.user.first_name}!
+                        </Row>
+                        <Row className="approved-hours">
+                            You have {Math.round(this.state.approved_hours*100)/100} approved hours.
+                        </Row>
+                        <Row className="pending-hours">
+                            You have {Math.round(this.state.pending_hours*100)/100} pending hours.
+                        </Row>
                     </Col>
                 </Row>
-                <Container className="progress-bar">
-                    <VProgressBar hours={this.state.hours}>
+                    <Container className="progress-container">
+                        <VProgressBar completeCount={this.state.approved_hours} pendingCount={this.state.pending_hours} className="progress-bar"/>
+                    </Container>
 
-                    </VProgressBar>
-                </Container>
                 <Row>
                     <Col align="center">
-                        <ReportHours/>
+                        <ReportHours onChange={()=>getHoursForStudent(this)}/>
```

Comment:
added this, it was recalling the state again and again.

Reply:
Thanks I was confused about the function of it